extensions/amp-viewer-integration/0.1/messaging/messaging.js

```diff
@@ -14,7 +14,20 @@
  * limitations under the License.
  */
 
-import {urls} from '../../../../src/config';
+/**
+ * Allows for runtime configuration. Internally, the runtime should
+ * use the src/config.js module for various constants. We can use the
+ * AMP_CONFIG global to translate user-defined configurations to this
+ * module.
+ * @type {!Object<string, string>}
+ */
+const env = self.AMP_CONFIG || {};
+
+const cdnProxyRegex =
+  (typeof env['cdnProxyRegex'] == 'string'
+    ? new RegExp(env['cdnProxyRegex'])
+    : env['cdnProxyRegex']) ||
+  /^https:\/\/([a-zA-Z0-9_-]+\.)?cdn\.ampproject\.org$/;
```

Comment:
I'm afraid this isn't acceptable for the CDN hosted script. We support compiling the v0+extensions using custom URLs (eg, bing can use their cache URLs instead of ours). If publishing this package outside of AMP's regular `gulp dist` has difficulties, then we need to fix the publishing script to support this.

Reply:
@rsimha This looks like an action item for you and team.

This extension now needs to be compiled before publishing a version.

Reply:
👍 (Could someone file a FR issue with the details?)

Reply:
Yep, I'll do it in a min.

Reply:
Filed https://github.com/ampproject/amphtml/issues/32110

Reply:
As an alternative (since this requires more work now), we could change the method to pass the regex as a new, optional param. So we'd pass the responsibility onto the caller to say "this domain is ok, too", instead of coding it into this script.

Reply:
So you mean passing the regex to `waitForHandshakeFromDocument(..., opt_cdnRegex)`, right? Yeah would work for us!